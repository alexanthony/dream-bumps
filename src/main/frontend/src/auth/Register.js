// @flow
import React, { PropTypes } from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router';
import {Field, reduxForm} from 'redux-form';
import {TextField, SelectField} from 'redux-form-material-ui';
import RaisedButton from 'material-ui/RaisedButton';
import {Card, CardTitle, CardText, CardActions} from 'material-ui/Card';
import {Cell} from 'material-grid/dist';
import MenuItem from 'material-ui/MenuItem';

import {signUp} from './actions';
import colleges from '../utils/colleges';

export const Register = ({handleSubmit}:{handleSubmit:Function}) => (
  <form onSubmit={handleSubmit}>
    <Cell col={6} offset={3}>
      <Card>
        <CardTitle title="Register"/>
        <CardText>
          <div>
            Already have an account? <Link to="/login">Login</Link>
          </div>
          <Field name="login" component={TextField} floatingLabelText="Username" fullWidth/>
          <Field name="email" component={TextField} floatingLabelText="E-Mail" fullWidth/>
          <Field name="password" component={TextField} floatingLabelText="Password" type="password" fullWidth/>
          <Field name="confirmPassword" component={TextField} floatingLabelText="Confirm Password" type="password" fullWidth/>
          <Field
            name="college"
            fullWidth
            floatingLabelText="College"
            component={SelectField}
          >
            {colleges.map(college =>
              <MenuItem
                key={college.value}
                value={college.value}
                primaryText={college.label}
              />
            )}
          </Field>
        </CardText>
        <CardActions>
          <RaisedButton primary label="Register" type="submit"/>
        </CardActions>
      </Card>
    </Cell>
  </form>
);

Register.propTypes = {
  handleSubmit: PropTypes.func
};

const validate = values => {
  if (values.password !== values.confirmPassword) {
    return {confirmPassword: 'Passwords must match.'};
  }
  return {};
}

const RegisterForm = reduxForm({
  form: 'register',
  initialValues: {
    langKey: 'en'
  },
  validate
})(Register);

export default connect(null, {onSubmit: signUp})(RegisterForm);
