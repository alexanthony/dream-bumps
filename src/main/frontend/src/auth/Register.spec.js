import React from 'react';
import { shallow } from 'enzyme';
import { Register } from './Register';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const component = shallow(
    <Register />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot();

});
