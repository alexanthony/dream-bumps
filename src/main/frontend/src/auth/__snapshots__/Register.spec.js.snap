exports[`test Renders properly 1`] = `
<form>
  <Cell
    col={6}
    offset={3}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardTitle
        title="Register" />
      <CardText>
        <div>
          Already have an account? 
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/login">
            Login
          </Link>
        </div>
        <Field
          component={[Function]}
          floatingLabelText="Username"
          fullWidth={true}
          name="login" />
        <Field
          component={[Function]}
          floatingLabelText="E-Mail"
          fullWidth={true}
          name="email" />
        <Field
          component={[Function]}
          floatingLabelText="Password"
          fullWidth={true}
          name="password"
          type="password" />
        <Field
          component={[Function]}
          floatingLabelText="Confirm Password"
          fullWidth={true}
          name="confirmPassword"
          type="password" />
      </CardText>
      <CardActions>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Register"
          labelPosition="after"
          primary={true}
          secondary={false}
          type="submit" />
      </CardActions>
    </Card>
  </Cell>
</form>
`;
